### PR TITLE
NPC alternate attack and friendly fire updates

### LINKED ISSUE #17240: NPCs using the Tazer and Tazer not knowing how to be used by non-Players
1. Give NPC a loaded Tazer.
2. Engage in combat.
3. Occasionally, be prompted shock something, even if NPC is far away.

Edit.
4. If NPC is engaged in combat and there are no enemies around the player, then prompted to shock something, you have no choice but to shock yourself to death.

### LINKED ISSUE #17258: NPC escape item and alternative attack item code terribly outdated
Causes #17240

NPCs use a hardcoded list of escape items and alternative attack items. This is obviously a horrible practice - it should be done at the very least with flags.

And it should be tested too. C4, mininuke and tazer are all player-only items at the moment.

### PR DESCRIPTION
**Pull request by Coolthulhu**
Reworked NPC alternate attack. Now it allows jsons to specify how should an item be used.
Fixes #17258
Fixes #17240

NPCs can activate and throw bombs and can throw items specified by json (currently combat knives and wooden spears, like in master, but also acid bombs). They can't use tazers.

NPC escape item usage removed. It didn't really work for a looooong time and it can't be fixed without a total rewrite.

Reworked friendly fire avoidance.
Instead of brute forcing some weird selection of points, the NPC will try to move in the direction that minimizes `rl_dist( pos(), target ) - rl_dist( pos(), friends )` where `friends` is the average of positions of all allies. This includes other NPCs.
Check for "can friendly fire" rewritten to be based on angles rather than brute forcing "thick lines" and checking if player is on them. Also includes allies.

Doesn't include vehicles as allies yet, but that wouldn't be hard to add now.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -5838,13 +5838,12 @@ std::string item::get_plant_name() const
...
-    // Note: doesn't handle the pipebomb or radio bombs
-    // Consider flagging dangerous items with an explicit flag instead
-    static const std::string explosion_string( "explosion" );
-    if( type->can_use( explosion_string ) ) {
+    if( has_flag( "DANGEROUS" ) ) {
         return true;
     }
 
+    // Note: Item should be dangerous regardless of what type of a container is it
+    // Visitable interface would skip some options
```
Comment (mugling) [inline src/item.cpp]: Perhaps it should skip them? In what circumstances would that be a problem? Hidden `contents` are effectively incorporated in their parent item and shouldn't be directly accessed. If you need access to one of those hidden items `item::is_dangerous` should contain an explicit check?

Comment (Coolthulhu) [inline src/item.cpp]: > In what circumstances would that be a problem?

Any case where contents are processed and could be dangerous.
Currently no such option is possible, but realistically:
* Grenade launcher magazine with a dud/activated time bomb
* Radio mine launcher with radio mine loaded into it. Or a magazine for it.
* Overheating battery
* Grenade stored in a sealed can

Comment (mugling) [inline src/item.cpp]: Those are all special cases and could be reasonably handled via `item::is_dangerous` - after all the magazine with a dud is equally as dangerous as the dud itself.

**Comment by mugling** [comment]
Needs rebase

**Comment by mugling** [comment]
This is quite difficult to test. We could do with a debug menu option for spawning a friendly NPC. The friendly fire checks work reasonably well but the usage of area effect weapons tends to be near-suicidal in most cases.

**Comment by Coolthulhu** [comment]
> This is quite difficult to test. We could do with a debug menu option for spawning a friendly NPC.

Mutate debug mind control. It allows you to talk to hostile NPCs and turn them friendly.
> the usage of area effect weapons tends to be near-suicidal in most cases.

Like in master, unfortunately. NPCs will need a kiting behavior before they can use those properly.

**Comment by Coolthulhu** [comment]
This PR would conflict with a lot of NPC code.

Is the non-suicidal grenade use mandatory? They're just as suicidal as in master. I can add some restrictions if it is needed, though.

**Comment by Coolthulhu** [comment]
17 days

Is there still something wrong with it?

**Pull request merged**